flink-streaming-connectors/flink-connector-cassandra/src/test/java/org/apache/flink/streaming/connectors/cassandra/CassandraConnectorITCase.java

```diff
@@ -318,6 +312,46 @@ protected void verifyResultsDataDiscardingUponRestore(
 		Assert.assertTrue("The following ID's were not found in the ResultSet: " + list.toString(), list.isEmpty());
 	}
 
+	@Override
+	protected void verifyResultsWhenScalingDown(CassandraTupleWriteAheadSink<Tuple3<String, Integer, Integer>> sink) throws Exception {
+
+		ArrayList<Integer> list = new ArrayList<>();
+		for (int x = 1; x < 34; x++) {
+			list.add(x);
+		}
+
+		ResultSet result = session.execute(SELECT_DATA_QUERY);
+		for (Row s : result) {
+			list.remove(new Integer(s.getInt("counter")));
+		}
+		Assert.assertTrue(list.isEmpty());
+	}
+
+	@Override
+	protected void verifyResultsWhenScalingUp(
+		CassandraTupleWriteAheadSink<Tuple3<String, Integer, Integer>> sink, int startElementCounter, int endElementCounter) {
+
+		// IMPORTANT NOTE:
+		//
+		// for cassandra we always have to start from 1 because
+		// all operators will share the same final db
+
+		ArrayList<Integer> list = new ArrayList<>();
+		for (int i = 1; i <= endElementCounter; i++) {
+			list.add(i);
+		}
+
+		ArrayList<Integer> expected = new ArrayList<>();
+		ResultSet result = session.execute(SELECT_DATA_QUERY);
+		for (Row s : result) {
+			expected.add(s.getInt("counter"));
+		}
+
+		Collections.sort(expected);
+
+		Assert.assertArrayEquals(list.toArray(), expected.toArray());
```

Comment:
why does this test follow a different approach to testing than `verifyResultsWhenScalingDown`?